Podcast ep11 (#92)
nextflow-io/website: content/podcast/2023/ep11_sarus_out_git.md

title=The Sarus container engine, staging outputs and the GITHUB_TOKEN
episode=11
description=The Sarus container engine, staging outputs and the GITHUB_TOKEN.
date=2023-01-25
type=podcast
subtype=News and Views
youtubeid=WHmRP9ixLqg
image=img/podcast_ep11.jpg
tags=news and views,opensource,community
status=published
author= Developer advocates
icon=logo_podcast_channels.jpg
~~~~~~

In this News and Views episode, [Chris Hakkaart](https://twitter.com/chris_hakk) and [Marcel Ribeiro-Dantas](https://twitter.com/mribeirodantas) discuss the hottest topics in the Nextflow world.

<!-- end-archive-description -->

#### The Sarus container engine

- Recently [Sarus](https://sarus.readthedocs.io/en/stable/) was [added to Nextflow](https://github.com/nextflow-io/nextflow/pull/3470) as a container engine.
- Sarus is an OCI-compatible container engine for HPC.
- Sarus is an alternative container runtime to Docker.
    - Sarus works by converting Docker images to a common format that can then be distributed and launched on HPC systems.
- Similar deployment options to other container engines using Nextflow. 

> #### New stageOutMode directives

Reviewer: ```suggestion
#### New stageOutMode options
```

> - Two new `stageOutMode` directive (`rclone` and `fcp`) have been added to Nextflow.

Reviewer: ```suggestion
- Two new `stageOutMode` options (`rclone` and `fcp`) have been added to Nextflow.
```

    - The `stageOutMode` directive defines how output files are staged-out from the scratch directory to the process work directory.
- [rclone](https://rclone.org/)
    - Has been described as "The Swiss army knife of cloud storage".
    - A command-line program to manage files on cloud storage.
    - Over 40 cloud storage products support `rclone`.
- [fcp](https://github.com/Svetlitski/fcp)
    - A significantly faster alternative to the classic Unix `cp` command.
    - Handle the most common use cases of cp with much higher performance.
    - `fcp` is optimized for systems with an SSD.

> - The new directives will improve options for file movement as they are staged from the scratch directory.

Reviewer: ```suggestion
- The new options will improve file movement as there are new ways for the files to be staged from the scratch directory in different circumstances.
```

> - The new options will improve file movement as there are new ways for the files to be staged from the scratch directory in different circumstances.

Reviewer: ```suggestion
- The new options will improve file management as there are new ways for the files to be staged from the scratch directory in different circumstances.
```

#### GITHUB_TOKEN

- Git has become the de-facto standard for source code version control systems.
- Nextflow provides built-in support for Git and the most popular Git hosting platforms.
- Nextflow does not require any special configuration to access public repositories but requires repository credentials to access private repositories.

> - The [recent Nextflow commit](https://github.com/nextflow-io/nextflow/commit/c4d39384c3536e6777bc0edcd9fd33f147aec462) adds the ability to use the `GITHUB_TOKEN` environmental variable.

Reviewer: ```suggestion
- The [recent Nextflow commit](https://github.com/nextflow-io/nextflow/commit/c4d39384c3536e6777bc0edcd9fd33f147aec462) adds the ability to use the `GITHUB_TOKEN` environment variable.
```

>     - The environmental variable is used when no creds are found in the source code management file.

Reviewer: ```suggestion
    - The environment variable is used when no credentials are found in the source code management configuration file (`$HOME/.nextflow/scm`).
```

>     - It is especially when accessing pipeline code from a GitHub action.

Reviewer: ```suggestion
    - It is especially useful when accessing pipeline code from a GitHub action.
```
